[AI Deploy] - speech to text app tuto

MathieuBsqt opened the pull request:
Create a new tutorial to deploy a Streamlit Speech to text application:

- doc created: `guide.en-gb.md `

- code reference: https://github.com/ovh/ai-training-examples/tree/main/apps/streamlit/speech-to-text 

- FR translations: 
Titre: **AI Deploy - Tutoriel - Créer et Déployer une Application de Speech to Text**
Extrait: **Comment créer et déployer une application de Speech to Text / Transcription audio **


Please, just wait for an AI Solutions team member review. Thanks!

MathieuBsqt committed:
[AI Deploy] - speech to text app tuto

eleapttn commented:
In the following command, please:
- remove the `./` at the beginning of the command
- avoid the use of your own github repo! You can probably add it in the `ai-training-examples`
- you should use the same Docker image for the job and for the app (soon, no more `one-for-all` image)

```console
./ovhai job run ovhcom/ai-training-one-for-all \
--cpu 12 \
--volume speech_to_text_app_models@GRA/:/workspace/speech_to_text_app_models:RW \
--volume https://github.com/MathieuBsqt/STT_APP_Models.git:/workspace/github_repo:rw \
-- bash -c 'pip install -r /workspace/github_repo/requirements.txt && python /workspace/github_repo/download_models.py'
``` 

To sum up, you should probably create an only one Docker image for the job and for the app. Add all the codes, etc on the same repo (`ai-training-examples`).

eleapttn commented:
I would have perhaps exchanged the two "blocks" (put the primary before) to avoid that the user connects a volume unnecessarily without having read the continuation.

```
ovhai app run \
      --default-http-port 8501 \
      --gpu 1 \
      --volume speech_to_text_app_models@GRA/:/workspace/models:RO \
      <shared-registry-address>/streamlit_app:latest
```

```
[!primary]

If you followed the optional part Import the models and save them locally, you can load the volume where your models are stored with the --volume parameter. This time, we put it in read-only (RO) mode because we only need to have an access to the models, so we can use them. We don't need to write or delete anything in this Object Storage container.

Otherwise, you can delete the --volume line, since it will not bring anything to your app.
```

eleapttn commented:
Perfect for the rest of the content, nice doc 😊 ✅ !

MathieuBsqt committed:
[AI Deploy] - speech to text app tuto update

eleapttn committed:
update image name in job command

Add the complete image name:
```
ovhai job run <shared-registry-address>/streamlit_app:latest \
    --cpu 12 \
    --volume speech_to_text_app_models@GRA/:/workspace/speech_to_text_app_models:RW \
    --volume https://github.com/ovh/ai-training-examples.git:/workspace/github_repo:rw \
    -- bash -c 'python /workspace/github_repo/apps/streamlit/speech-to-text/download_models.py'
```

eleapttn reviewed (approved):
All is good on my side!

Y0Coss committed:
Proofreading AI Deploy tuto Streamlit Speech-to-text

OVH-Team-Guides commented on pages/platform/ai/deploy_tuto_09_streamlit_speech_to_text_app/guide.en-gb.md:
```diff
@@ -0,0 +1,246 @@
+
+## Objective
+
+The purpose of this documentation is to **Deploy the Speech to Text Application** we have realised in the blog articles using [Streamlit](https://streamlit.io/) and pre-trained models. 
```
Hello @MathieuBsqt , could you please add a link to the mentioned blog article ?

MathieuBsqt replied:
Hi @OVH-Team-Guides , thanks, I just corrected this. 
I couldn't do it before because the blog post wasn't out yet.

MathieuBsqt committed:
Update guide.en-gb.md

add the link to the mentioned blog articles

Y0Coss committed:
Minor proofreading

Y0Coss committed:
Duplicate AI Deploy tuto Streamlit STT

Commit:
PUSH Duplicate AI Deploy tuto Streamlit STT

Commit:
Duplicate AI Deploy tuto Streamlit STT

Commit:
PUSH Duplicate AI Deploy tuto Streamlit STT

Y0Coss committed:
FR versions fixes

OVH-Team-Guides commented:
Thank you for your contribution @MathieuBsqt
